### PR TITLE
wx_compat is no more.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

wxpython4 is required for python3 support, so wx_compat can be removed, hem, deprecated.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -383,8 +383,7 @@ def grab_frame(self, **savefig_kwargs):
                              dpi=self.dpi, **savefig_kwargs)
         except (RuntimeError, IOError) as e:
             out, err = self._proc.communicate()
-            _log.info('MovieWriter -- Error '
-                           'running proc:\n%s\n%s' % (out, err))
+            _log.info('MovieWriter -- Error running proc:\n%s\n%s', out, err)
```
Comment (jenshnielsen) [inline lib/matplotlib/animation.py]: This change does not seem to belong in this pr?

Comment (anntzer) [reply lib/matplotlib/animation.py]: forgot to undo that; reverted that part.
